### PR TITLE
feat: add kill count notifier

### LINKED ISSUE #1: Notification for Boss KC
Settings would probably allow custom x KC between notifications, or 50,100,250,500,1000 

image

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #1

### CODE REVIEW

File: src/main/java/dinkplugin/DinkPluginConfig.java
Comment (iProdigy) [inline src/main/java/dinkplugin/DinkPluginConfig.java:505]: Let me know what the default values for these config settings should be (likely needs to be revised)

File: src/main/java/dinkplugin/DinkPluginConfig.java
Comment (Felanbird) [inline src/main/java/dinkplugin/DinkPluginConfig.java:506]: swap these 2 so killCountMessage is last

**Review by Felanbird — COMMENTED** [review]
image

Tested with a KC interval of 2 - Screenshot setting does not work

File: src/main/java/dinkplugin/notifiers/KillCountNotifier.java
Comment (Felanbird) [inline src/main/java/dinkplugin/notifiers/KillCountNotifier.java:54]: image

**Comment by Felanbird** [comment]
```
Your subdued Wintertodt count is: 359 
Your Barrows chest count is: 268
```

File: src/main/java/dinkplugin/notifiers/KillCountNotifier.java
Comment (pajlada) [reply src/main/java/dinkplugin/notifiers/KillCountNotifier.java:54]: I think the functionality should stay but as an option

File: src/main/java/dinkplugin/notifiers/KillCountNotifier.java
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/KillCountNotifier.java:71]: I like this way of exposing testing the KC_REGEX without exposing the regex itself and then returning a more strict type

**Comment by iProdigy** [comment]
For some boss kills, felan pointed out the kill time & PB are sent in chat: https://cdn.discordapp.com/attachments/901356769371107398/1032366619789369364/collectionImage.png

So, a future feature could also trigger a notification on new PB, even if interval isn't satisfied

This increases the complexity of the notifier and the message format is different for group boss fights, so I have not yet implemented this

File: src/main/java/dinkplugin/notifiers/KillCountNotifier.java
Comment (iProdigy) [inline src/main/java/dinkplugin/notifiers/KillCountNotifier.java:64]: This approach doesn't work for Wintertodt, Barrows, or the raids. For Wintertodt, I can hardcode the howling storm gif, if desired.

File: src/main/java/dinkplugin/DinkPluginConfig.java
Comment (Felanbird) [inline src/main/java/dinkplugin/DinkPluginConfig.java:483]:
```suggestion
        return false;
```
Should be disabled by default to be inline with the other notifs, also an issue with our speedRun notification but that's another PR.

File: src/main/java/dinkplugin/DinkPluginConfig.java
Comment (Felanbird) [inline src/main/java/dinkplugin/DinkPluginConfig.java:494]:
```suggestion
        return false;
```

**Review by Felanbird — APPROVED** [review]
Functionality works as expected 👍 tested the following scenarios:

"Normal" boss (Vorkath) at an interval of 1
Special boss cases Wintertodt & Barrows at an interval of 1
Always-post-first-notif while using an interval of 50
Vorkath when Send Image is disabled (an image of Vorkath's model is posted in the discord notif)

Barrows & Wintertodt with send image disabled was also tested, but as noted above is not currently handled.

Only thing that could require some testing is raids notifications, but that'll require some effort to test.

**Review by pajlada — APPROVED** [review]
Looks good to me :+1: 

In the conflict merging, make sure your changelog entry is under the `## Unreleased` section

I explicitly made a release without this functionality even though it looks good & seems to work as expected to lessen the reviewer load on [the Plugin Hub PR](https://github.com/runelite/plugin-hub/pull/3497)

I'll let @Mm2PL have the last review/approval before it finally gets merged but I don't foresee any issues

**Review by Mm2PL — APPROVED** [review]

**Comment by Mm2PL** [comment]
Will merge in as soon as raids are tested :)

**Comment by Felanbird** [comment]
> Will merge in as soon as raids are tested :)

(we're not going to test raids unless pajlada can wrangle notheowner dfantasy and 🅱️ upinic)

**Pull request merged**
